code/datums/disease.dm

```diff
@@ -121,10 +121,11 @@ var/list/diseases = typesof(/datum/disease) - /datum/disease
 		check_range = 1 // everything else, like infect-on-contact things, only infect things on top of it
 
 	if(isturf(source.loc))
-		for(var/mob/living/carbon/M in oview(check_range, source))
-			if(isturf(M.loc))
-				if(AStar(source.loc, M.loc, /turf/proc/AdjacentTurfs, /turf/proc/Distance, check_range))
-					M.contract_disease(src, 0, 1, force_spread)
+		for(var/mob/living/carbon/victim in oview(check_range, source))
+			if(isturf(victim.loc))
+				if(AStar(source.loc, victim.loc, /turf/proc/AdjacentTurfs, /turf/proc/Distance, check_range) && !locate(src) in victim.viruses)
+					if(get_infection_chance(victim))
+						victim.contract_disease(src, 0, 1, force_spread)
```

Comment:
contract disease should be checking for all this in line 81

[passed = check_disease_pass_clothes(target_zone)](https://github.com/cmss13-devs/cmss13/blob/ab44c0dd2a9ea424cb7585d20d5d0905ee9a537d/code/datums/diseases/mob_procs.dm#L81)

and it does it a little neater with a variable on clothes, is that not working correctly?

Reply:
You get infected almost instantly without these bits of code, but I will double check.

Reply:
I guess I can nuke my changes and beat this to work.